i18n-helpers/src/bin/markdown-gettext.rs

```diff
@@ -0,0 +1,248 @@
+use anyhow::{anyhow, bail, Context};
+use mdbook_i18n_helpers::translate;
+use polib::catalog::Catalog;
+use polib::po_file;
+use std::fs::File;
+use std::io::{BufReader, Read, Write};
+use std::path::PathBuf;
+use std::{env, fs};
+
+fn split_argument(arg: &String) -> Option<String> {
+    let splitted: Vec<_> = arg.split('=').collect();
+    if splitted.len() == 2 {
+        return Some(splitted[1].to_string());
+    }
+
+    None
+}
+
+fn build_catalog(lang_option: Option<String>) -> anyhow::Result<Catalog> {
+    match lang_option {
+        Some(lang_path) => {
+            let pot_path = PathBuf::from(&lang_path);
+
+            if pot_path.exists() {
+                let catalog = po_file::parse(&pot_path)
+                    .map_err(|err| anyhow!("{err}"))
+                    .with_context(|| format!("Could not parse {:?} as PO file", pot_path))?;
+
+                return Ok(catalog);
+            }
+
+            bail!("--po must be specified")
+        }
+        None => bail!("--po must be specified"),
+    }
+}
+
+fn translate_files(
+    catalog: Catalog,
+    files: Vec<PathBuf>,
+    output_path: PathBuf,
+) -> anyhow::Result<()> {
+    for file_path in files.iter() {
+        let content = File::open(&file_path)
+            .with_context(|| format!("Failed to open file: {}", file_path.display()))?;
+        let mut buf_reader = BufReader::new(content);
+        let mut contents = String::new();
+        buf_reader
+            .read_to_string(&mut contents)
+            .with_context(|| format!("Failed to read file: {}", file_path.display()))?;
+
+        let translated_content = translate(&contents, &catalog);
+
+        let output_file_name = file_path.file_name().ok_or(anyhow!("Invalid file name"))?;
+        let output_file_path = output_path.join(output_file_name);
+
+        let mut output_file = File::create(&output_file_path)
+            .with_context(|| format!("Failed to create file: {}", output_file_path.display()))?;
+
+        output_file
+            .write_all(translated_content.as_bytes())
+            .with_context(|| format!("Failed to write to file: {}", output_file_path.display()))?;
+    }
+
+    Ok(())
+}
+
+fn validate_file(path: &PathBuf) -> bool {
+    path.is_file() && path.extension().map_or(false, |ext| ext == "md")
+}
+
+fn allocate_files(
+    dir_option: Option<String>,
+    file_option: Option<String>,
+) -> anyhow::Result<Vec<PathBuf>> {
+    let mut valid_files = Vec::new();
+
+    match (dir_option, file_option) {
+        (Some(_), Some(_)) => return bail!("Only one of --dir or --f should be specified"),
+        (None, None) => return bail!("Either --dir or --f must be specified"),
+        (Some(dir_path), None) => {
+            let full_dir = PathBuf::from(&dir_path);
+            if !full_dir.exists() || !full_dir.is_dir() {
+                return bail!("Directory does not exist: {}", full_dir.display());
```

Comment:
[`bail!`](https://docs.rs/anyhow/latest/anyhow/macro.bail.html) already has a `return` inside it:

```suggestion
                bail!("Directory does not exist: {}", full_dir.display());
```